NebulaPatcher/Patches/Transpilers/PowerSystem_Transpiler.cs

```diff
@@ -15,142 +17,196 @@ namespace NebulaPatcher.Patches.Transpilers;
+            .Insert(
+                new CodeInstruction(OpCodes.Ldarg_0),
+                nodeIdCI,
+                coreEnergyRatioCI,
+                new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(PowerSystem_Transpiler), nameof(SetChargerRequriePower))),
+                new CodeInstruction(OpCodes.Br_S, label)
+            );
+
+        // Check if chargers are local before adding the energy to the mecha
+        // from: if (this.nodePool[num75].id == num75)
+        // get:  num75 (nodeId)
+        codeMatcher.End()
+            .MatchBack(true,
+                new CodeMatch(OpCodes.Ldfld, AccessTools.Field(typeof(PowerNodeComponent), nameof(PowerNodeComponent.id))),
+                new CodeMatch(OpCodes.Ldloc_S));
+        if (codeMatcher.IsInvalid)
+        {
+            Log.Error("PowerSystem_GameTick_Transpiler 3 failed. Mod version not compatible with game version.");
+            return codeInstructions;
+        }
+        var loadNodeIdInstruction = codeMatcher.InstructionAt(0);
+
+        /*
+        from:
+            if (num77 <= 0 || entityAnimPool[entityId5].state != 2U)
+        to:
+            if (num77 <= 0 || entityAnimPool[entityId5].state != 2U || !IsNotLocal(this, num75))
+        */
+        codeMatcher.MatchForward(true,
+                new CodeMatch(OpCodes.Ldfld, AccessTools.Field(typeof(AnimData), nameof(AnimData.state))),
+                new CodeMatch(OpCodes.Ldc_I4_2),
+                new CodeMatch(OpCodes.Bne_Un));
+        if (codeMatcher.IsInvalid)
+        {
+            Log.Error("PowerSystem_GameTick_Transpiler 4 failed. Mod version not compatible with game version.");
+            return codeInstructions;
+        }
+
+        codeMatcher
+             .Insert(
+                new CodeInstruction(OpCodes.Ldarg_0),
+                loadNodeIdInstruction,
+                new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(PowerSystem_Transpiler), nameof(IsNotLocal))),
+                new CodeInstruction(OpCodes.Or)
+            );
+
+        return codeMatcher.InstructionEnumeration();
+    }
 
-        codeMatcher = codeMatcher
-            .Advance(1)
-            .InsertAndAdvance(new CodeInstruction(OpCodes.Ldarg_0))
-            .InsertAndAdvance(new CodeInstruction(OpCodes.Ldloc_S, 59))
-            .InsertAndAdvance(new CodeInstruction(OpCodes.Ldloc_S, 22))
-            .Insert(HarmonyLib.Transpilers.EmitDelegate<PlayerChargesAtTower>((_this, powerNodeId, powerNetId) =>
+    private static bool IsNotLocal(PowerSystem powerSystem, int nodeId)
+    {
+        if (!Multiplayer.IsActive)
+            return false;
+
+        if (GameMain.mainPlayer.planetId != powerSystem.factory.planetId)
+            return true;
+
+        return !Multiplayer.Session.PowerTowers.LocalChargerIds.Contains(nodeId);
+    }
+
+#pragma warning disable CA1868
+    private static void SetChargerRequriePower(PowerSystem powerSystem, int nodeId, float coreEnergyRatio)
```

Comment:
```suggestion
    private static void SetChargerRequirePower(PowerSystem powerSystem, int nodeId, float coreEnergyRatio)
```

Or should it be Requires/Required?

Reply:
Given that it is updating `PowerNodeComponent.requiredEnergy`, Required may be better.